### PR TITLE
Add support for enabling Github's "require resolved conversations" option

### PR DESCRIPTION
**Pull request by sfc-gh-kleonhard**
## Hey, I just made a Pull Request!

This PR plumbs through the Github option to require all conversations be resolved before merging a PR.

API reference: https://octokit.github.io/rest.js/v19

The API's default is false, as is the default set in this PR

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: .changeset/healthy-shrimps-notice.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-scaffolder-backend': minor
+---
+
+Add option to require Github conversation resolution
```
Comment (jhaals) [inline .changeset/healthy-shrimps-notice.md:5]:
```suggestion
Added `requireConversationResolution` template option to `github:repo:create`, `github:repo:push` and `publish:github`
```

File: plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/github.test.ts [hunk 2]
```diff
@@ -997,6 +1010,7 @@ describe('publish:github', () => {
       requireCodeOwnerReviews: false,
       requiredStatusCheckContexts: [],
       requireBranchesToBeUpToDate: true,
+      requiredConversationResolution: false,
```
Comment (jhaals) [inline plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/github.test.ts:1013]: Let's have at least one test in each of these actions that use the new option :)

File: plugins/scaffolder-backend/src/scaffolder/actions/builtin/helpers.ts [hunk 3]
```diff
@@ -237,6 +239,7 @@ export const enableBranchProtectionOnDefaultRepoBranch = async ({
           bypass_pull_request_allowances: bypassPullRequestAllowances,
           dismiss_stale_reviews: dismissStaleReviews,
         },
+        required_conversation_resolution: requiredConversationResolution,
```
Comment (jhaals) [inline plugins/scaffolder-backend/src/scaffolder/actions/builtin/helpers.ts:242]:
```suggestion
        require_conversation_resolution: requiredConversationResolution,
```
What do you think about naming it Require Conversation Resolution instead? I think it read a bit better.

**Review by jhaals — CHANGES_REQUESTED** [review]
Thanks for the PR! I've added some suggestions and questions :)

File: plugins/scaffolder-backend/src/scaffolder/actions/builtin/helpers.ts (hunk 3 above)
Comment (sfc-gh-kleonhard) [reply plugins/scaffolder-backend/src/scaffolder/actions/builtin/helpers.ts:242]: I totally agree that it reads better, but this name matches the underlying [API](https://github3.readthedocs.io/en/latest/api-reference/repos.html#github3.repos.branch.BranchProtection.required_conversation_resolution). The line suggested on must remain `required_conversation_resolution`, that's the name used in the client.

I could rename it everywhere else though. WDYT?

File: .changeset/healthy-shrimps-notice.md (hunk 1 above)
Comment (sfc-gh-kleonhard) [reply .changeset/healthy-shrimps-notice.md]: Done.

File: plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/github.test.ts (hunk 2 above)
Comment (sfc-gh-kleonhard) [reply plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/github.test.ts:1013]: Done.

File: plugins/scaffolder-backend/src/scaffolder/actions/builtin/helpers.ts (hunk 3 above)
Comment (jhaals) [reply plugins/scaffolder-backend/src/scaffolder/actions/builtin/helpers.ts:242]: I discussed it with the other maintainers and we opted to keep it `required_conversation_resolution` as it's closer to the github API 👍

File: .changeset/healthy-shrimps-notice.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-scaffolder-backend': minor
+---
+
+Added `requireConversationResolution` template option to `github:repo:create`, `github:repo:push` and `publish:github`
```
Comment (jhaals) [inline .changeset/healthy-shrimps-notice.md:5]:
```suggestion
Added `requiredConversationResolution` template option to `github:repo:create`, `github:repo:push` and `publish:github`
```

Let's keep the name you first suggested :)

File: plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/github.test.ts (hunk 2 above)
Comment (jhaals) [reply plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/github.test.ts:1013]: Hmm, are you sure? :) I was expecting at least one test in each asserting `requiredConversationResolution: true`

**Review by jhaals — CHANGES_REQUESTED** [review]

File: plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/github.test.ts (hunk 2 above)
Comment (sfc-gh-kleonhard) [reply plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/github.test.ts:1013]: Whoops! Rookie mistake; I didn't actually push the new tests! PTAL

File: plugins/scaffolder-backend/src/scaffolder/actions/builtin/helpers.ts (hunk 3 above)
Comment (sfc-gh-kleonhard) [reply plugins/scaffolder-backend/src/scaffolder/actions/builtin/helpers.ts:242]: Thanks for the quick turnaround!

File: .changeset/healthy-shrimps-notice.md (hunk 4 above)
Comment (sfc-gh-kleonhard) [reply .changeset/healthy-shrimps-notice.md]: Done.

**Review by jhaals — APPROVED** [review]
Thank you! 👏

**Pull request merged**